ddtrace/contrib/internal/langgraph/patch.py

```diff
@@ -98,6 +98,87 @@ async def traced_runnable_seq_ainvoke(langgraph, pin, func, instance, args, kwar
     return result
 
 
+@with_traced_module
+def traced_runnable_seq_astream(langgraph, pin, func, instance, args, kwargs):
+    """
+    Typically, RunnableSeq.ainvoke() is called when being run as a task for its parent Pregel (CompiledGraph).
+    However, when using Pregel.astream_events() (inherited from langchain's Runnable), it calls RunnableSeq.astream()
+    instead.
+
+    This function returns a generator wrapper that yields the results of RunnableSeq.astream(),
+    ending the span after the stream is consumed, otherwise following the logic of traced_runnable_seq_ainvoke().
+    """
+    integration: LangGraphIntegration = langgraph._datadog_integration
+
+    node_name = _get_node_name(instance)
+
+    if node_name in ("_write", "_route"):
+        return func(*args, **kwargs)
+    if node_name == "LangGraph":
+        config = get_argument_value(args, kwargs, 1, "config", optional=True) or {}
+        config.get("metadata", {})["_dd.subgraph"] = True
+        return func(*args, **kwargs)
```

Comment:
Since we have multiple repetitions of this block, can we try extracting this into a helper method?